src/tests/CoreMangLib/system/delegate/miscellaneous/SecondLevelIndirectionWithNullThis.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System;
+using System.Reflection;
+using Xunit;
+
+public class Program
+{
+    public int value = 100;
+
+    public void Update(int num)
+    {
+        value += num;
+    }
+
+    [Fact]
+    public static int TestEntryPoint()
+    {
+        Program prog = new Program();
+
+        Action<int> action = prog.Update;
+        Action<int> secondLevel = action.Invoke;
+
+        FieldInfo targetField = typeof(Delegate).GetField("_target", BindingFlags.Instance | BindingFlags.NonPublic);
```

Comment:
If somebody reported a bug that required use of private reflection like this to repro, we would resolve it as won't fix.

Is there a way to hit the null reference exception via supported public surface? (ie without use of private reflection on runtime internals)

Reply:
When constructing delegate directly, `Delegate.Invoke` is instance method and the compiler selected ctor checks for null target.

```csharp
            var invokeMethod = typeof(Action).GetMethod("Invoke");
            var wrappedDelegate = invokeMethod.CreateDelegate<Action<Action>>();
            wrappedDelegate(null);
```

This code seems reproducing. Creating a static delegate for instance method allows the instance to be passed in. It's not a virtual method so the shuffling stub should not be checking for null. It's public reflection and should be supported.

Reply:
Or, maybe using IL to invoke the delegate constructor with `.ctor(null, methodof(Action.Invoke))`.